### PR TITLE
fix(#831, #832, #834, #851): unify Shape's deep-copy docs, boolean delegation, bounds hygiene, and point-classification bridge

### PR DESCRIPTION
**Pull request by gsdali**
## What & why

Four duplication-audit findings from #382/Pass 2a, combined into one PR because together they
interlink `Shape.swift`, `Shape+Topology.swift`, `Shape+Analysis.swift`, and `Shape+Modeling.swift`
— splitting them would mean two PRs touching the same files concurrently (see the parent task for
the exact cross-reference matrix).

**#831 — pure documentation fix, no behavior change.** `Shape` has three "deep copy" APIs:
`copy(copyGeometry:copyMesh:)` (`BRepBuilderAPI_Copy`), static `Shape.deepCopy(_:copyGeometry:copyMesh:)`
(`BRepTools_CopyModification` — the same underlying mechanism `copy()` uses, reached directly instead
of through that convenience wrapper), and instance `deepCopy()` (`TNaming_CopyShape::CopyTool`).
`docs/thread-safety.md` and `docs/naming-conventions.md` both claimed the no-argument instance
`deepCopy()` uses `BRepBuilderAPI_Copy` with independent geometry — wrong on both counts, confirmed
by reading `TNaming_TranslateTool::UpdateFace`/`UpdateEdge` (`Libraries/occt-src/...`): it assigns
the *same* `Geom_Surface`/`Geom_Curve`/`Poly_Triangulation` handles to the copy, cloning topology
only. Fixed both docs, added a comparison table, and cross-referenced all three methods' own doc
comments in `Shape+Topology.swift` so the distinction is visible at the call site, not just in
`docs/`. **Flagged, not fixed**: `Shape.isSelfIntersecting(hardTimeout:)` relies on the no-argument
`deepCopy()` for thread isolation in a background-timeout pattern — since that copy shares geometry
handles (and their mutable evaluation caches) with the original, this is a well-evidenced latent
risk, documented in both `docs/thread-safety.md` and the method's own doc comment as a candidate
follow-up. Not changed here, per the parent task's explicit instruction not to alter
`isSelfIntersecting`'s behavior in this PR.

**#832 — genuine bug fix (missing timeout protection), one measured non-finding.**
`Shape+Modeling.swift`'s `fused/subtracted/intersected(tolerance:/glue:)` (six entry points,
pre-#202/#206, v0.114.0 era) called their own narrow bridge functions directly, bypassing the
`defaultBooleanTimeout` (120s) watchdog `Shape.union/subtracting/intersection` have carried since
#206 specifically because a self-intersecting/inside-out operand can hang `BRepAlgoAPI_Cut`
indefinitely. Fixed by making all six delegate to the canonical family internally — same signature,
same return type, zero API break, now inheriting the watchdog. `GlueMode` (`Shape+Modeling.swift`,
`shift=0/full=1/off=2`) and `BooleanGlue` (`Shape.swift`, `off=0/shift=1/full=2`) encode the same
`BOPAlgo_GlueEnum` choice with different raw-value orderings; no caller anywhere in the tree uses
raw values (confirmed by grep), so kept both declarations (case names match, unlike #834's tuple
problem) with an explicit case-name mapping (`GlueMode.asBooleanGlue`, `internal` so the test can
assert it directly) and cross-referencing doc comments warning against conflating them by raw value.
**Measured, not assumed**: the audit also flagged the old code's *unconditional* `SetFuzzyValue`
call (vs. `union`'s `if fuzzyValue > 0.0`) as forwarding a negative tolerance into OCCT untested. A
small-gap fixture built specifically to discriminate "ignored" from "applied" (posFuzzy closes the
gap, tolerance:0 doesn't) showed a literal `-0.01` behaves identically to `tolerance: 0` on **both**
the pre- and post-delegation code — `BRepAlgoAPI_Fuse::SetFuzzyValue` already discards a negative
value internally. So this part of the audit's concern doesn't correspond to an observable bug;
recorded as a correction in the test's own comment rather than left as an unverified claim.
**Review follow-up (this update):** a reviewer confirmed a real tradeoff the first pass introduced —
these six entry points had no wall-clock bound at all before this PR, so a caller whose
fuzzy-tolerance boolean on a large/complex assembly previously took, say, 150s and eventually
succeeded would now silently get `nil` at the inherited 120s default instead, with no way to
override since none of the six exposed a `timeout:` parameter. Removing the timeout isn't the fix —
it exists precisely because #206 needs it — so all six now also take a `timeout:` parameter
(default `Shape.defaultBooleanTimeout`, same type/semantics as `union`/`subtracting`/`intersection`'s
own `timeout:`, `0`/negative = unbounded). This is an additive parameter with a default value, so
every existing call site keeps compiling unchanged; confirmed, not just asserted, by building the
full package against the new signatures with zero source changes required elsewhere in the tree.

**#834 — genuine bug (confirmed, not fully fixable without a SemVer break), fixed to the extent safe
in this PR.** `Shape.bounds` (`Shape.swift`, `OCCTShapeGetBounds`) and `Shape.boundingBox`
(`Shape+Analysis.swift`, `OCCTShapeBoundingBox`) compute the identical `Bnd_Box`/`BRepBndLib::Add`,
but only `boundingBox` guards the void case explicitly (`Bnd_Box::IsVoid()` before `Get()`).
`bounds` relied on `Bnd_Box::Get()`'s `Standard_ConstructionError` throw + a catch-all zeroing every
output — same *output* for a void shape, but exception-driven rather than deliberate, and unlike
`boundingBox`'s `Optional`, `bounds`'s non-optional tuple return type has **no way** to signal "no
geometry", so a void shape's `(0,0,0)-(0,0,0)` is indistinguishable from a genuine zero-size shape
at the origin. Confirmed with a new test (`voidShapeBoundingBoxIsNilButBoundsFabricatesZero`) using
a genuinely void shape (`Shape.compound([])` refuses to construct — count must be ≥1 — so the test
uses a far-disjoint `intersection`, empirically confirmed via `boundingBox == nil`). Fixed the
bridge hygiene (`OCCTShapeGetBounds` now guards `IsVoid()` explicitly, matching the deliberate
convention its two siblings already use, zero output change). **Not fixed**: making `bounds`
Optional would be the correct fix but is a breaking signature change affecting 20 call sites across
the tree (a mix of tests and two production consumers) — flagged here rather than executed silently,
per the parent task's explicit branch for this case. Both production consumers
(`DrawingAutoDimensions.addAutoDimensions`, `ThreadFeatures.buildThreadedRodDirect`) were
investigated and found to already degrade safely without a guard (documented inline at each site) —
so the fabrication is a real, live divergence with no test coverage, but not (currently) a silent
wrong-output bug at either flagged call site. Documented the divergence in `bounds`'/`size`'/`center`'s
doc comments, `Wire.bounds`'s doc comment (it forwards to `Shape.bounds`), and the two
`docs/reference/` entries (`Shape-Features.md`, `Document-Transforms.md`).

**#851 — genuine dedup + doc fix.** `Shape.classify(point:tolerance:)` (`Shape+Analysis.swift`,
`OCCTClassifyPointInSolid`, `BRepClass3d_SolidClassifier`) and `Shape.classifyPoint(_:tolerance:)`
(`Shape+Topology.swift`, `OCCTShapeClassifyPoint`) answered the identical question via two
independently-written bridge implementations — the latter hand-built
`BRepClass3d_SolidExplorer` + `BRepClass3d_SClassifier`, the exact pair `BRepClass3d_SolidClassifier`
wraps internally (confirmed by reading its header). `docs/reference/Document-Math-Bounds.md`
already (mis)attributed `classifyPoint` to `BRepClass3d_SolidClassifier` — wrong before this PR,
true after it: rather than fix the doc to match the old (redundant) implementation, fixed the
implementation to match the doc, since that's the real dedup. `OCCTShapeClassifyPoint` now
constructs `BRepClass3d_SolidClassifier` directly and reads its result through the same
`mapTopAbsState()` helper `OCCTClassifyPointInSolid` already used, instead of a separate raw
`(int32_t)` cast that happened to agree only because `TopAbs_State`'s ordinals lined up. Zero
behavior change (verified: the new `pointOnBoxFace` test passes, and the ordinal mapping is
unchanged). `PointState` (`.on`) and `PointClassification` (`.onBoundary`) share raw values
case-for-case but not case *names* — unlike #850's `Face.SurfaceType`/`Surface.SurfaceType` (where
the case sets were identical and a typealias was a pure zero-behavior-change merge), a typealias
here would silently rename one side's case and break source compatibility. Kept as two declarations,
cross-referenced in both doc comments. Also closed the coverage gap the audit found: `classifyPoint`'s
`.on` boundary case had no test anywhere in the tree (`classify`'s did) — added
`BRepClass3dTests.pointOnBoxFace`, mirroring the existing `PointClassificationTests.pointOnBoxFace`
exactly.

## Which parts are pure dedup vs. genuine bug fixes

| Issue | Pure dedup | Genuine bug found & fixed | Flagged, not fixed |
|---|---|---|---|
| #831 | doc/comment unification | — (docs were wrong, now correct; no code behavior changed) | `isSelfIntersecting(hardTimeout:)`'s latent geometry-sharing risk |
| #832 | six entry points now share one bridge code path | missing timeout watchdog (real, now fixed, now overridable via `timeout:`); GlueMode/BooleanGlue raw-value mismatch (real, cross-referenced not merged) | — |
| #834 | — | void-shape fabrication is real and newly tested; bridge hygiene fixed | `bounds`'s non-optional signature can't signal void without a SemVer break |
| #851 | bridge implementation unified onto one OCCT mechanism | doc misattribution (fixed by making it true); missing `.on` boundary test (added) | — |

## Test results

- `swift build` — clean.
- `swift build --target OCCTModelingTests` / `OCCTTopologyTests` / `OCCTAnalysisTests` — clean.
- Full `OCCTModelingTests`: 645/645 passed (was 644; +1 for the new review-follow-up test).
- Full `OCCTTopologyTests`: 477/477 passed.
- Full `OCCTAnalysisTests`: 530/530 passed.
- New `Issue832BooleanDelegationTests` (5 tests, incl. the new `explicitTimeoutIsThreadedThrough`
  added for the review follow-up above): all pass.
- New `voidShapeBoundingBoxIsNilButBoundsFabricatesZero` (#834): passes.
- New `BRepClass3dTests.pointOnBoxFace` (#851): passes.
- Full `swift test` (~4900+ tests across all domain targets): run, results in the PR thread /
  reported by the agent.
- All 6 static gate scripts (`check-bridge-index`, `check-null-handle-guards`,
  `check-docs-defaults`, `check-docs-existence`, `derive-bridge-header-split --verify`,
  `count-operations`) — clean, 0 findings.
- All gate `--self-test`s (5 gates + census + changelog-transcription) — all cases pass.
- `census-unmeasured-values.py` — no new findings in any touched file.

**Prove-the-test-fails, run for every new test** (per `okf/policies/prove-the-test-fails.md`):
- `voidShapeBoundingBoxIsNilButBoundsFabricatesZero`: injected (`boundingBox`'s void guard disabled)
  → failed with the expected assertion mismatch; restored → passed.
- `BRepClass3dTests.pointOnBoxFace`: injected (`classifyPoint` hardcoded to `.outside`) → failed
  (3 tests broke, as expected); restored → passed.
- `Issue832BooleanDelegationTests.glueModeMapsByCaseName`: injected (`asBooleanGlue` reverted to a
  raw-value cast) → failed on the direct mapping assertions (the volume-parity half of the same test
  did **not** catch it — recorded in the test's own comment as a real measurement, not an assumption,
  since glue mode has no observable effect on a simple coincident-face fixture); restored → passed.
- `Issue832BooleanDelegationTests.negativeToleranceIgnored`: injected (reverted `fused(with:tolerance:)`
  to its pre-#832 direct bridge call) → still passed, which is itself the finding recorded above (no
  observable difference — OCCT already ignores negative fuzzy values).
- `Issue832BooleanDelegationTests.explicitTimeoutIsThreadedThrough` (review follow-up): injected
  (one delegation call, e.g. `fused(with:tolerance:timeout:)`, changed to call `union(other,
  fuzzyValue: tolerance)` without forwarding `timeout:`) → failed (the tiny-timeout assertion for
  that entry point no longer returned `nil`); restored → passed. Confirms `timeout:` is actually
  threaded through to the underlying `union`/`subtracting`/`intersection` call, not merely accepted
  and ignored.

### `Shape`'s deep-copy, boolean, bounds, and point-classification duplicates unified (#831, #832, #834, #851)

- **#831**: Fixed `docs/thread-safety.md` and `docs/naming-conventions.md`, which both incorrectly
  claimed the no-argument `shape.deepCopy()` uses `BRepBuilderAPI_Copy` with independent geometry.
  It actually uses `TNaming_CopyShape::CopyTool` and only clones topology — `Geom_Surface`/
  `Geom_Curve`/`Poly_Triangulation` handles are shared with the original. For genuinely independent
  geometry (e.g. for concurrent use on separate threads), use `copy(copyGeometry: true)` or the
  static `Shape.deepCopy(_:copyGeometry:copyMesh:)` instead. Flagged (not changed):
  `Shape.isSelfIntersecting(hardTimeout:)`'s use of the no-argument `deepCopy()` for background-thread
  isolation is a latent risk under this corrected understanding.
- **#832**: `Shape+Modeling`'s `fused/subtracted/intersected(tolerance:/glue:)` now delegate to
  `Shape.union/subtracting/intersection`. **These six entry points are now timeout-bounded by
  default (120s, `Shape.defaultBooleanTimeout`) — previously they had no wall-clock bound at all**,
  so a caller relying on an operation that legitimately runs longer than 120s will now get `nil`
  instead of eventually succeeding, unless it opts out. Each of the six gained a `timeout:`
  parameter (default `Shape.defaultBooleanTimeout`, same type/semantics as `union`'s own
  `timeout:`; `0` or negative disables the bound, matching the old unbounded behavior) — an
  additive parameter with a default value, so existing call sites keep compiling unchanged.
- **#834**: `OCCTShapeGetBounds` (backing `Shape.bounds`) now guards `Bnd_Box::IsVoid()` explicitly
  before `Get()`, matching `Shape.boundingBox`'s established convention, instead of relying on
  exception unwinding. Output is unchanged. Documented (not changed, pending a SemVer-registered
  signature change) that `Shape.bounds`/`size`/`center` fabricate `(0,0,0)`/`.zero` for a void shape,
  indistinguishable from real zero-size geometry — `Shape.boundingBox` is the void-safe alternative.
- **#851**: `Shape.classifyPoint(_:tolerance:)`'s bridge implementation now routes through
  `BRepClass3d_SolidClassifier` directly, matching `Shape.classify(point:tolerance:)`'s mechanism
  instead of hand-building the lower-level `BRepClass3d_SolidExplorer`/`BRepClass3d_SClassifier`
  pair that class already wraps. No behavior change. Added missing test coverage for the `.on`
  boundary case on `classifyPoint`.

## SemVer impact

PATCH, re-examined after the review follow-up above rather than left as the original,
pre-follow-up classification. #831, #834, and #851 are unchanged from the original reasoning:
documentation-only, an internal implementation delegation with identical signature and
identical/strictly-safer behavior, or a bridge-hygiene change with byte-identical output. #832 now
includes a genuinely new public surface — six added `timeout:` parameters — so it's worth stating
explicitly why that's still PATCH rather than MINOR: every one of the six is an **additive
parameter with a default value**, so no existing call site's source changes, no signature is
removed or narrowed, and the default (`Shape.defaultBooleanTimeout`) preserves the new watchdog
protection rather than silently reverting to the old unbounded behavior. That matches this
project's own "tiny additive feature = patch bump" convention. No case removed or renamed, no
existing default changed. The one place a real fix would require a breaking change — making
`Shape.bounds` `Optional` (#834) — was deliberately **not** executed in this PR; it is proposed in
the PR description above as a candidate MINOR/MAJOR change for a future PR, not shipped here.

## Checklist

- [x] New or changed behavior is covered by a unit test in the same PR.
- [x] Every new test was run once with its subject broken, and the failure is reported above.
- [x] The CHANGELOG entry above is complete, and `docs/CHANGELOG.md` is **not** in this diff.
- [x] The SemVer impact above is stated, and `docs/SEMVER.md` is **not** in this diff.

## Notes for the reviewer

- This branch targets `refactor/382-pass2a`, not `main`, per the parent audit issue (#382). The
  `Closes #831` / `#832` / `#834` / `#851` lines below will not auto-close on merge into that
  branch — expected, per #382's own process.
- #834's proposed `Shape.bounds` → `Optional` signature change is deliberately **not** in this diff.
  It's the one finding in this batch where the "genuine bug fix" and "zero API break" goals
  conflict; per the parent task's explicit instruction, that conflict is surfaced here for a human
  decision rather than resolved unilaterally. The 20 existing call sites (mostly test fixtures, plus
  `DrawingAutoDimensions.swift`/`ThreadFeatures.swift` in production code) were read in full; both
  production sites already degrade safely without a guard (documented inline), so the risk today is
  "wrong answer looks like a valid one for a caller nobody's written yet," not an observed bug.
- #832's `negativeToleranceIgnored` test is kept as a parity/regression check even though the
  measurement showed no old-vs-new behavioral difference — recorded plainly per
  `okf/policies/measure-dont-assume.md` rather than silently dropping the test or overstating the
  fix.
- Addressed the review comment on the original `fused(with:tolerance:)` diff hunk: added
  `timeout:` (default `Shape.defaultBooleanTimeout`) to all six `fused`/`subtracted`/
  `intersected(tolerance:/glue:)` entry points, updated the CHANGELOG entry and SemVer section
  above to state the behavior change plainly instead of leaving it doc-comment-only, and added
  `explicitTimeoutIsThreadedThrough` (proved to fail against a reverted delegation call before the
  fix, per `okf/policies/prove-the-test-fails.md`).

Closes #831
Closes #832
Closes #834
Closes #851

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: Sources/OCCTSwift/Shape+Modeling.swift [hunk 1]
```diff
@@ -2997,46 +2997,86 @@ extension Shape {
...
+    ///
+    /// - Note: Delegates to ``union(_:fuzzyValue:glue:timeout:)`` (#832), so this now carries the
+    ///   same ``defaultBooleanTimeout`` (120s) watchdog and the same "negative tolerance is
+    ///   ignored" contract as the canonical boolean family — neither of which this narrower,
+    ///   pre-#202/#206 entry point had on its own. Same signature, same return type; only the
+    ///   underlying safety changed.
     public func fused(with other: Shape, tolerance: Double) -> Shape? {
-        guard let ref = OCCTBooleanFuseWithTolerance(handle, other.handle, tolerance) else { return nil }
-        return Shape(handle: ref)
+        union(other, fuzzyValue: tolerance)
```
[the remark this answers is not in the document]
Comment (gsdali) [reply Sources/OCCTSwift/Shape+Modeling.swift]: Fixed in 48f98b7: added a `timeout:` parameter (default `Shape.defaultBooleanTimeout`, same type/semantics as `union(_:fuzzyValue:glue:timeout:)`'s own `timeout:`) to all six entry points — `fused/subtracted/intersected(tolerance:)` and `fused/subtracted/intersected(glue:)` — forwarded through to the underlying `union`/`subtracting`/`intersection` call. Default keeps the #832 protection (no silent regression); a caller whose operation legitimately needs longer than 120s can now override it, matching your suggestion rather than removing the timeout outright (which would reopen the #206 hang risk this whole delegation exists to close).

Additive parameter with a default value, so existing call sites keep compiling unchanged — confirmed via a clean full `swift build`, not just asserted.

Added `Issue832BooleanDelegationTests.explicitTimeoutIsThreadedThrough`, which proves `timeout:` actually reaches the underlying call (not just accepted and ignored): a tiny deadline (1e-7s) returns `nil` for all six entry points, and a sane one (60s) still succeeds with the correct volume. Proved the test fails first, per this repo's policy: reverted one delegation call to drop the `timeout:` forward, confirmed the test failed on exactly that assertion, restored, confirmed it passed again.

Also updated the CHANGELOG entry (in the PR body, per this repo's changelog-on-merge policy) to state the behavior change plainly — these six are now timeout-bounded by default where they were previously unbounded, with a `timeout:` opt-out — instead of leaving it doc-comment-only, and updated `docs/reference/Document-Mesh-Fixing.md` for the six new signatures (caught by `check-docs-defaults.py`, now clean).

Re-examined the SemVer classification given the new parameter rather than leaving the original PATCH claim unexamined: still PATCH, since every change is an additive parameter with a default value (no existing call site changes, no default's behavior changes) — matches this project's "tiny additive feature = patch bump" convention. Said so explicitly in the PR body's SemVer section rather than assuming it.

`OCCTModelingTests`: 645/645 passed (was 644, +1 for the new test). All 6 static gate scripts clean.
